gsy_framework/forward_markets/aggregated_ssp_profile.py

```diff
@@ -0,0 +1,165 @@
+        ).range("hours", 1)
+
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        return float(self._SSP_AGGREGATED_PROFILE[timeslot.format("M-H")])
+
+
+class WeeklyAggregatedSSPProfile(AggregatedSSPProfileBase):
+    """Return weekly-aggregated profile of the SSP."""
+
+    SSP_AGGREGATED_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/daily.csv"
+
+    def _get_timeslots(self, start_time: DateTime, end_time: DateTime) -> Iterable:
+        assert end_time - start_time >= duration(weeks=1), "Time period should be >= 1 week."
+        return period(
+            start_time.start_of("week"),
+            end_time.start_of("week") - duration(weeks=1)
+        ).range("weeks", 1)
+
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        return sum([
+            float(self._SSP_AGGREGATED_PROFILE[str(t.month)])
+            for t in period(timeslot, timeslot.add(days=6)).range("days", 1)
+        ])
+
+
+class MonthlyAggregatedSSPProfile(AggregatedSSPProfileBase):
+    """Return monthly-aggregated profile of the SSP."""
+    SSP_AGGREGATED_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/non_leap_monthly.csv"
+    LEAP_YEAR_SSP_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/leap_monthly.csv"
+
+    def _get_timeslots(self, start_time: DateTime, end_time: DateTime) -> Iterable:
+        assert end_time - start_time >= duration(months=1), "Time period should be >= 1 month."
+        return period(
+            start=start_time.start_of("month"),
+            end=end_time.start_of("month") - duration(days=1)
+        ).range("months", 1)
+
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        if timeslot.is_leap_year():
+            return float(self._LEAP_YEAR_SSP_AGGREGATED_PROFILE[timeslot.format("M")])
+        return float(self._SSP_AGGREGATED_PROFILE[timeslot.format("M")])
+
+
+class YearlyAggregatedSSPProfile(AggregatedSSPProfileBase):
+    """Return yearly-aggregated profile of the SSP."""
+    SSP_AGGREGATED_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/non_leap_yearly.csv"
+    LEAP_YEAR_SSP_AGGREGATED_PROFILE_PATH = RESOURCES_PATH / "aggregated_ssp/leap_yearly.csv"
+
+    def _get_timeslots(self, start_time: DateTime, end_time: DateTime) -> Iterable:
+        assert end_time - start_time >= duration(years=1), "Time period should be >= 1 year."
+        return period(
+            start=start_time.start_of("year"),
+            end=end_time.start_of("year") - duration(days=1),
+        ).range("years", 1)
+
+    def _get_timeslot_energy_kWh(self, timeslot: DateTime) -> float:
+        if timeslot.is_leap_year():
+            return float(self._LEAP_YEAR_SSP_AGGREGATED_PROFILE[""])
+        return float(self._SSP_AGGREGATED_PROFILE[""])
+
+
+def get_aggregated_SSP_profile(
+        capacity_kWh: float, start_time: DateTime,
+        end_time: DateTime, resolution: AggregationResolution
+):
```

Comment:
Nit: I'd suggest staying consistent with the rest of the codebase and put the `):` at the end of the previous line:
```suggestion
def get_aggregated_SSP_profile(
        capacity_kWh: float, start_time: DateTime,
        end_time: DateTime, resolution: AggregationResolution):
```